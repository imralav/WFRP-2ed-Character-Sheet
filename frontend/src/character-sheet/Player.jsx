import React from 'react';
import {FormattedMessage} from 'react-intl';

export default function Player(props) {
    const handleChange = event => {
       const changeData = {name: event.target.name, value: event.target.value};
       props.onChange(changeData);
    };
    return (
        <table>
            <thead>
            <tr>
                <th colSpan="2" className="header-font-size">
                    <FormattedMessage id="player" defaultMessage="Player"/>
                </th>
            </tr>
            </thead>
            <tbody>
            <tr>
                <td>
                    <div>
                        <label className="label-font-size">
                            <FormattedMessage id="name" defaultMessage="Name"/>:
                        </label>
                        <input type="text" value={props.data.name} name="name" onChange={event => handleChange(event)} className="content-font-size"/>
                    </div>
                </td>
                <td className="w-40p">
                    <div>
                        <label className="label-font-size">
                            <FormattedMessage id="game-master" defaultMessage="Game Master"/>:
                        </label>
                        <input type="text" value={props.data.gameMaster} name="gameMaster" onChange={event => handleChange(event)} className="content-font-size"/>
                    </div>
                </td>
            </tr>
            <tr>
                <td>
                    <div>
                        <label className="label-font-size">
                            <FormattedMessage id="campaign" defaultMessage="Campaign"/>:
                        </label>
                        <input type="text" value={props.data.campaign} name="campaign" onChange={event => handleChange(event)} className="content-font-size"/>
                    </div>
                </td>
                <td>
                    <div>
                        <label className="label-font-size">
                            <FormattedMessage id="campaign-year" defaultMessage="Campaign Year"/>:
                        </label>
                        <input type="text" value={props.data.campaignYear} name="campaignYear" onChange={event => handleChange(event)} className="content-font-size"/>
                    </div>
                </td>
            </tr>
            </tbody>
        </table>
    );
}