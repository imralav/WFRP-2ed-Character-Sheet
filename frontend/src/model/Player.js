import simplePropertyReplacementReducer from "./SimplePropertyReplacementReducer";

const DEFAULT_PLAYER = {name: '',gameMaster: '',campaign: '',campaignYear: ''};
export const UPDATE_PLAYER_ACTION = 'UPDATE_PLAYER';
export default simplePropertyReplacementReducer(UPDATE_PLAYER_ACTION, DEFAULT_PLAYER);