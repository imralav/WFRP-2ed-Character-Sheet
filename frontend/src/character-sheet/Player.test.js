import React from 'react';
import Player from './Player';
import renderer from 'react-test-renderer';

import en from 'react-intl/locale-data/en';
import {addLocaleData, IntlProvider} from 'react-intl';
addLocaleData(en);

describe('Player', () => {
    let component;
    let componentRoot;
    let characterData;
    let tree;
    let onChangeFunction;
    beforeEach(() => {
        characterData = {
            name: 'someName',
            gameMaster: 'someGameMaster',
            campaign: 'someCampaign',
            campaignYear: 'someCampaignYear'
        };
        onChangeFunction = jest.fn();
        component = renderer.create(
            <IntlProvider locale="en" messages={{}}>
                <Player data={characterData} onChange={changeData => onChangeFunction(changeData)}/>
            </IntlProvider>
        );
        componentRoot = component.root;
        tree = component.toJSON();
    });
    afterEach(() => {
        characterData = null;
        component = null;
        componentRoot = null;
        tree = null;
    });
    it('should match previous snapshot', () => {
        expect(tree).toMatchSnapshot();
    });
    ['name', 'gameMaster', 'campaign', 'campaignYear'].forEach(propertyName => {
        let input;
        beforeEach(() => {
            input = componentRoot.findByProps({name: propertyName});
        });
        afterEach(() => {
            input = null;
        });
        it(`should have correct ${propertyName}`, () => {
            expect(input.props.value).toEqual(characterData[propertyName]);
        });
        describe('on changed input value', () => {
            const event = {
                target: {
                    name: 'eventName',
                    value: 'someValue'
                }
            };
            beforeEach(() => {
                input.props.onChange(event);
            });
            it('should call function from props with correct arguments', () => {
                expect(onChangeFunction).toHaveBeenCalledWith({name: event.target.name, value: event.target.value});
            });
        });
    });
});
