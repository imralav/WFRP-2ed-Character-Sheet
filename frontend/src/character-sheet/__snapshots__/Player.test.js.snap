// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should match previous snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th
        colSpan="2"
      >
        <span>
          Player
        </span>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <div>
          <label>
            <span>
              Name
            </span>
            :
          </label>
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value="someName"
          />
        </div>
      </td>
      <td
        className="w-40p"
      >
        <div>
          <label>
            <span>
              Game Master
            </span>
            :
          </label>
          <input
            name="gameMaster"
            onChange={[Function]}
            type="text"
            value="someGameMaster"
          />
        </div>
      </td>
    </tr>
    <tr>
      <td>
        <div>
          <label>
            <span>
              Campaign
            </span>
            :
          </label>
          <input
            name="campaign"
            onChange={[Function]}
            type="text"
            value="someCampaign"
          />
        </div>
      </td>
      <td>
        <div>
          <label>
            <span>
              Campaign Year
            </span>
            :
          </label>
          <input
            name="campaignYear"
            onChange={[Function]}
            type="text"
            value="someCampaignYear"
          />
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
